### PR TITLE
create-app: Add "yarn dev" command to templated new apps

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Adds the "yarn dev" command (and it's corresponding `concurrently` package) to the templated apps to simplify local development.

Also bumps concurrently in the main app, a bump which has very few minor features and the reason for the major bump is dropped Node 8 support.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/create-app/templates/default-app/package.json.hbs [hunk 1]
```diff
@@ -29,9 +30,10 @@
   },
   "devDependencies": {
     "@backstage/cli": "^{{version '@backstage/cli'}}",
-    "@spotify/prettier-config": "^7.0.0",
+    "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
-    "prettier": "^1.19.1"
+    "prettier": "^1.19.1",
+    "@spotify/prettier-config": "^7.0.0"
```
Comment (freben) [inline packages/create-app/templates/default-app/package.json.hbs:36]: heh, why was this moved :) no biggie, just curious

File: .changeset/few-suns-cheer.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/create-app': patch
+---
+
+Adds "yarn dev" command to simplify local development.
```
Comment (freben) [inline .changeset/few-suns-cheer.md:5]: On create-app changes, we usually add a few lines to inform users how they can make the corresponding change to their local `packages/app`.

File: packages/create-app/templates/default-app/package.json.hbs (hunk 1 above)
Comment (adamdmharvey) [reply packages/create-app/templates/default-app/package.json.hbs:36]: so they are alphabetical? happy to move it back up 😁

Comment (freben) [reply packages/create-app/templates/default-app/package.json.hbs:36]: yepyep, doesn't make a difference except next time someone does a `yarn add` the script will re-sort so the diffs are unnecessarily larger 🧶

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by timbonicus** [comment]
Nice addition, @adamdmharvey! 🙌  I was wanting this recently myself. Mind updating the [getting-started docs](https://github.com/backstage/backstage/blob/master/docs/getting-started/create-an-app.md#run-the-app) to reflect this too?

**Review by Rugvip — APPROVED** [review]
👍  :pray: :shipit:

File: packages/create-app/templates/default-app/package.json.hbs (hunk 1 above)
Comment (adamdmharvey) [reply packages/create-app/templates/default-app/package.json.hbs]: ah - sorry, didn't know `@` auto sorted earlier in the path. tested offline, and learned something. fixed!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by adamdmharvey** [comment]
@timbonicus Good idea! Did a minor tweak to the doco, what do you think?

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by adamdmharvey** [comment]
Assume all the failures in CI are unrelated; some seem to infer problems with an Azure testing component, etc.

**Comment by freben** [comment]
soon to be rebaseable upon #4866

**Comment by benjdlambert** [comment]
#4866 is merged, go ahead and rebase now @adamdmharvey :)

**Comment by adamdmharvey** [comment]
@freben @benjdlambert sorry to ask, but any suggestions on exactly what rebase command to use so i don't bork my fork? (ha it rhymes) Last I did a `git pull upstream master` I pulled in a bunch of unsigned commits which then blocked me for DCO. What approach would you use in this instance to rebase?

**Comment by freben** [comment]
yeah don't use the two-argument `pull`. I think this is the most straight forward and easy-to-grok sequence:

```
# make sure your master is new and fresh
git checkout master
git pull
# go back to your branch and rebase it
git checkout concurrently3
git rebase master
```

Now, of course, things can go wrong in a rebase. There may be merge issues it tells you to fix (`git status` will actually be very informative about things at that point). But best case - and commonly - it goes just fine.

**Comment by adamdmharvey** [comment]
hmm.  yeah i dont see how that's different, other than a `git pull upstream master` forcing an automatic merge upon rebase, versus yours just rewrites a fork?  so it worked, but reflects my commits twice it seems, and the merge commit itself isn't signed off so DCO isn't happy.  grumble grumble.  ha!  need to play around a lot more with this whole sign off stuff.

**Comment by adamdmharvey** [comment]
And maybe it's the fast forward which threw me...

```
backstage on  concurrently3 [$?⇕] via ⬢ v12.16.2 took 4s
❯ git push
To github.com:adamdmharvey/backstage
 ! [rejected]            concurrently3 -> concurrently3 (non-fast-forward)
error: failed to push some refs to 'git@github.com:adamdmharvey/backstage'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

backstage on  concurrently3 [$?⇕] via ⬢ v12.16.2
❯ git pull
Merge made by the 'recursive' strategy.

backstage on  concurrently3 [$?⇡] via ⬢ v12.16.2
❯ git push
Counting objects: 35, done.
Delta compression using up to 12 threads.
Compressing objects: 100% (35/35), done.
Writing objects: 100% (35/35), 3.71 KiB | 3.71 MiB/s, done.
Total 35 (delta 24), reused 0 (delta 0)
remote: Resolving deltas: 100% (24/24), completed with 10 local objects.
To github.com:adamdmharvey/backstage
   b371c0f8e..edd1f0dc7  concurrently3 -> concurrently3
```

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by freben** [comment]
Well, so, `pull` is devious, it does much more than you think. :) You can look at it as a form of macro. I try to do everything I can to make it work in fast-forward mode only. Essentially you could go all the way to doing a **fetch** of the upstream and resetting master to the new value of upstream/master...

I think your usage of `pull` is what bites you overall!

You tried to do a push which the server rejected (correctly), because your branch was not able to be applied with a fast-forward. What that means is, what you had locally wasn't a simple delta on top of the branch tip on the upstream. And that's entirely to be expected, precisely because of the rebase. It made your local branch utterly incompatible with the remote, so what needed to follow was a **force push** (`git push -f`). This of course sounds scary and should not be used on branches where several people work in parallel, and this is one of the arguments people have against its usage.

Starting state (letters are commits on a row):

```
server has: A - B
you have:   A - B
```

Then you performed a rebase.

```
server has: A - B
you have:   X - Y
```

To your eyes they (`A` and `X` for example) have the same contents, but they have entirely different hashes and hang off of a different part of the tree.

The ONLY way that the server would have permitted a push, would be if you had e.g. `A - B - C`. Then it could just advance a pointer up to `C` and be done. That's the fast forward mode. But your commits look completely alien to it so it says no.

Then you did a `pull` instead. That was the mistake that led to duplicates. Now git tries to do this big scary macro operation of fetching `A - B` from the remote, and hammering them "somehow" into your `X - Y`. You may find that it placed a confusing merge commit into the mix and there seem to be duplicates of stuff.

```
server has: A - B
you have:   X - Y - NOISE
```

If you had done a force push instead:

```
server has: X - Y
you have:   X - Y
```

so it would mirror your local state.

What makes it all the more confusing is that there are actually even more branches at play. `upstream/master`, `upstream/concurrently3`, `master`, `concurrently3` all play into things :)

**Comment by adamdmharvey** [comment]
Appreciate you taking the time with that explanation, @freben !  The force isn't something I've ever really grok'd. I think this may clean it up so let's see!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4846-seou7i7inzjvkjnkp93v2lhelqjrzx8y.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60429be165f817688503adf3

**Comment by adamdmharvey** [comment]
Yay went green; going to merge in and we can iterate as necessary.
